transport/http/peer_test.go

```diff
@@ -90,16 +90,25 @@ func TestHTTPWithRoundRobin(t *testing.T) {
 	temporary.Stop()
 	// One of these requests may fail since one of the peers has gone down but
 	// the HTTP transport will not know until a request is attempted.
-	integrationtest.Call(ctx, c)
-	integrationtest.Call(ctx, c)
+	err1 := integrationtest.Call(ctx, c)
+	err2 := integrationtest.Call(ctx, c)
+	if err1 == nil && err2 == nil {
+		t.Fatal("both calls succeeded after bringing one peer down")
+	}
+	if err1 != nil && err2 != nil {
+		t.Fatal("neither call succeeded after bringing one peer down")
+	}
 	// All subsequent should succeed since the peer should be removed on
 	// connection fail.
 	integrationtest.Blast(ctx, t, c)
 
 	// Restore the server on the temporary port.
-	restored, _ := spec.NewServer(t, temporaryAddr)
-	defer restored.Stop()
-	integrationtest.Blast(ctx, t, c)
+	// TODO the http client unfortunately reuses idle connections from the
+	// prior generation of the http server on this port, rendering this test
+	// useless.
```

Comment:
I *believe* the fix is to set the [KeepAlive](https://godoc.org/go.uber.org/yarpc/transport/http#KeepAlive) and [MaxIdleConnsPerHost](https://godoc.org/go.uber.org/yarpc/transport/http#MaxIdleConnsPerHost) to 0 but I could be wrong.

Reply:
I'll leave this for a future PR, it's a quick fix

Reply:
Zeroing KeepAlive and MaxIdleConnsPerHose did not have the expected effect. Some requests would still go to the old server. Landing this until we can resolve the mystery.